Sources/Core/Request/FormURLEncoder.swift

```diff
@@ -44,7 +44,7 @@ extension CharacterSet {
     static let urlFormAllowed: CharacterSet = {
         var allowed = CharacterSet.urlQueryAllowed
         allowed.insert(" ")
-        allowed.remove(charactersIn: "+/?")
+        allowed.remove(charactersIn: "+/?&")
```

Comment:
I think we should change the `CharacterSet` to be 
```swift
CharacterSet(charactersIn: "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-._* ")
```

So that we're more explicit by what's allowed - reference [here](https://url.spec.whatwg.org/#urlencoded-serializing)

Reply:
Note the space isn't actually allowed, but it already has the special handling to replace with "+" above.

Reply:
Updated